floodlight/models/kinematics.py

```diff
@@ -0,0 +1,102 @@
+import warnings
+from dataclasses import dataclass
+
+import numpy as np
+
+from floodlight.utils.types import Numeric
+from floodlight import Pitch, XY
+from floodlight.core.property import PlayerProperty
+
+
+@dataclass
+class ModelDistance:
+    """Class for calculating distances of players on the pitch.
+
+    Parameters
+    __________
+    pitch: Pitch
+        Pitch object of the data. Should at least include information about the data's
+        unit. If pitch.unit is 'percent', length and width of the pitch are also needed.
+
+    """
+
+    pitch: Pitch = None
+    length: Numeric = None
+    width: Numeric = None
+    unit: str = None
+
+    def __init__(self, pitch):
+        self.unit = pitch.unit
+        if pitch.length is not None:
+            self.length = pitch.length
+        if pitch.width is not None:
+            self.width = pitch.width
+        if pitch.unit is not None:
+            self.unit = pitch.unit
+        else:
+            warnings.warn(
+                "No information about pitch unit. Coordinates are assumed to be in m!"
+            )
+        self._distance_euclidean = None
+        self.framerate = None
+
+    def __str__(self):
+        return (
+            f"Floodlight ModelDistance Class with unit [{self.unit}],"
+            f"length {self.length} and width {self.width}"
+        )
+
+    def fit(self, xy: XY, direction: str = None):
```

Comment:
I'm not sure if "direction" is an ideal name, plus I think "xy" is a sensible default

Reply:
Changed "xy" to plane. Is there already a convention on how we call the XY object as an argument?

"direction" is a bit tricky. It is used in the Pitch object documentation in the same context but I agree that it isn't ideal. I would suggest something similar to, but not "axis", as it's taken by numpy. Maybe we can decide later with the other models as I suggest we will have the same optionality there.

Reply:
Sorry for the confusion, I think calling the XY object "xy" is fine. I meant that ``direction`` is most likely to be "xy" rather than "x" or "y", so it could be chosen as a default (instead of None).

I agree on your point concerning the name of ``direction``.  One option would be to use ``dim`` in analogy to numpy, but not sure if thats really a good idea.